apollo-federation/src/schema/mod.rs

```diff
@@ -214,6 +217,17 @@ impl FederationSchema {
             None => Ok(None),
         }
     }
+
+    pub(crate) fn is_fed_2(&self) -> bool {
+        self.metadata()
+            .and_then(|metadata| metadata.for_identity(&Identity::federation_identity()))
+            .map_or(false, |federation| {
```

Comment:
Very minor suggestion (and only recently available in our toolchain):
```suggestion
            .is_some_and(|federation| {
```